flink-streaming-java/src/main/java/org/apache/flink/streaming/api/datastream/ConnectedStreams.java

```diff
@@ -234,62 +235,62 @@ public StreamExecutionEnvironment getExecutionEnvironment() {
 	}
 
 	/**
-	 * Applies the given {@link TimelyCoFlatMapFunction} on the connected input streams,
+	 * Applies the given {@link CoProcessFunction} on the connected input streams,
 	 * thereby creating a transformed output stream.
 	 *
 	 * <p>The function will be called for every element in the streams and can produce
 	 * zero or more output. The function can also query the time and set timers. When
 	 * reacting to the firing of set timers the function can emit yet more elements.
 	 *
-	 * <p>A {@link org.apache.flink.streaming.api.functions.co.RichTimelyCoFlatMapFunction}
+	 * <p>A {@link RichCoProcessFunction}
 	 * can be used to gain access to features provided by the
 	 * {@link org.apache.flink.api.common.functions.RichFunction} interface.
 	 *
-	 * @param coFlatMapper The {@link TimelyCoFlatMapFunction} that is called for each element
+	 * @param coProcessFunction The {@link CoProcessFunction} that is called for each element
 	 *                      in the stream.
 	 *
-	 * @param <R> The of elements emitted by the {@code TimelyCoFlatMapFunction}.
+	 * @param <R> The of elements emitted by the {@code CoProcessFunction}.
 	 *
 	 * @return The transformed {@link DataStream}.
 	 */
-	public <R> SingleOutputStreamOperator<R> flatMap(
-			TimelyCoFlatMapFunction<IN1, IN2, R> coFlatMapper) {
+	public <R> SingleOutputStreamOperator<R> process(
+			CoProcessFunction<IN1, IN2, R> coProcessFunction) {
 
-		TypeInformation<R> outTypeInfo = TypeExtractor.getBinaryOperatorReturnType(coFlatMapper,
-				TimelyCoFlatMapFunction.class, false, true, getType1(), getType2(),
+		TypeInformation<R> outTypeInfo = TypeExtractor.getBinaryOperatorReturnType(coProcessFunction,
+				CoProcessFunction.class, false, true, getType1(), getType2(),
 				Utils.getCallLocationName(), true);
 
-		return flatMap(coFlatMapper, outTypeInfo);
+		return process(coProcessFunction, outTypeInfo);
 	}
 
 	/**
-	 * Applies the given {@link TimelyCoFlatMapFunction} on the connected input streams,
+	 * Applies the given {@link CoProcessFunction} on the connected input streams,
 	 * thereby creating a transformed output stream.
 	 *
 	 * <p>The function will be called for every element in the streams and can produce
 	 * zero or more output. The function can also query the time and set timers. When
 	 * reacting to the firing of set timers the function can emit yet more elements.
 	 *
```

Comment:
<p>The function will be called for every element in the input streams and can produce zero or more
	 * output elements. Contrary to the {@link #flatMap(CoFlatMapFunction)} function, this function can also
	 * query the time and set timers. When reacting to the firing of set timers the function can directly
	 * emit elements and/or register yet more timers.